### ISSUE TITLE
Updating master.cf fails on Centos Stream 8 on initial setup

### ISSUE THREAD

**Issue by sazzle2611** [bodgit/puppet-postfix#16]
We have just reinstalled a few of our servers with Centos Stream 8 and the initial puppet run fails with these lines for every config

```
Notice: /Stage[main]/Postfix::Config/Postfix_master[anvil/unix]/ensure: created
Error: /Stage[main]/Postfix::Config/Postfix_master[anvil/unix]: Could not evaluate: Failed to save Augeas tree to file. See debug logs for details.
Notice: /Stage[main]/Postfix::Config/Postfix_master[bounce/unix]/ensure: created
Error: /Stage[main]/Postfix::Config/Postfix_master[bounce/unix]: Could not evaluate: Failed to save Augeas tree to file. See debug logs for details.
```

We found the workaround was to truncate the master.cf file and then run puppet again and it worked perfectly.

This is not a massive issue for us as we have used the workaround but thought you'd like to know and might help someone else.

herculesteam/augeasproviders_core version 2.6
puppet version 7.5.0

Thanks,
Sarah

**Comment by bodgit** [comment]
This is actually a bug in Augeas which is used under the hood to manipulate the `master.cf` file. It's choking on the service `postlog/unix-dgram` as it doesn't recognise the `-dgram` bit.

This is fixed in the Augeas source already however they've not made a new release in over two years so consequently it's not made it into a Puppet agent release yet.

I think I can include the fixed lens in this module and it will override the bundled one.

**Comment by bonivart** [comment]
That sounds good to me.

**Comment by bodgit** [comment]
I've published v3.1.0 which adds support for EL8 and installs the updated lens.
